dvc/config.py

```diff
@@ -24,13 +24,29 @@ def __init__(self, msg):
 
 
 class NoRemoteError(ConfigError):
-    def __init__(self, command):
-        msg = (
-            "no remote specified. Setup default remote with\n"
-            "    dvc remote default <name>\n"
-            "or use:\n"
-            "    dvc {} -r <name>\n".format(command)
-        )
+    def __init__(self, command, *, config=None):
+        has_any_remote = False
+        if config:
+            has_any_remote = bool(
+                config.list_options(
+                    Config.SECTION_REMOTE_REGEX, Config.SECTION_REMOTE_URL
+                )
+            )
+
+        if has_any_remote:
+            msg = (
+                "no remote specified. Setup default remote with\n"
+                "    dvc remote default <remote name>\n"
+                "or use:\n"
+                "    dvc {} -r <remote name>".format(command)
+            )
+        else:
+            msg = (
+                "no remote specified. Create a remote with\n"
+                "    dvc remote add <remote name> <remote url>\n"
```

Comment:
This part is no relevan without `-d`, because if someone would've provided the remove, we would go `if remote` branch of execution (see below where you raise). So the hint here should assume `-d`.

Reply:
That's right!